src/main/appender.cpp

```diff
@@ -375,6 +377,31 @@ void Appender::FlushInternal(ColumnDataCollection &collection) {
 	context->Append(*description, collection);
 }
 
+void Appender::AppendDefault() {
+	if (!defaults[column]) {
+		throw InvalidInputException("Failed to append DEFAULT, this column does not have a DEFAULT value");
+	}
+	auto &default_expr = *defaults[column];
+	if (!default_expr.IsScalar()) {
+		throw InvalidInputException("Only columns with simple DEFAULT values are supported");
+	}
+
+	auto default_copy = default_expr.Copy();
+
+	auto &type = types[column];
+	auto binder = Binder::CreateBinder(*context);
+	ConstantBinder default_binder(*binder, *context, "DEFAULT value");
+	default_binder.target_type = type;
+	auto bound_default = default_binder.Bind(default_copy);
+
+	Value result_value;
+	if (!ExpressionExecutor::TryEvaluateScalar(*context, *bound_default, result_value)) {
```

Comment:
For a performance optimization - if the expression has `IsFoldable()`, we can cache the `result_value`. This should greatly improve performance in the general case (as this can essentially just become `AppendValue`).

Reply:
For another performance optimization, we could delay populating the chunk with defaults until FlushChunk is called, we then perform a masked ExecuteExpression into the chunk

One issue with this could be that things like `nextval` are not called when the user expects it though
appending to two tables interleaved that both target the same sequence for example

Reply:
That's fine. We don't offer any guarantees around Appenders aside from when they are flushed anyway.

I think the bigger issue is that there is no guarantee that `AppendDefault` is called for the same columns for every row - although you could detect that and bail-out of this optimization of course.

Reply:
I was thinking of using a selection vector / validity mask to mask out the rows that AppendDefault was not used on
I believe I saw similar behavior for filters, though I would have to look it up in the code.

In any case it will be faster than doing an ExpressionExecutor::Execute for a single row every time AppendDefault is called for non foldable default expressions.

Reply:
I might do that later though, or add a FIXME
for now we run the ExpressionExecutor::Execute every time AppendDefault is called when we could not fold the expression at Appender creation time.